packages/apps/MintingB2B/src/pages/NFTDetails/index.tsx

```diff
@@ -139,6 +141,28 @@ const WalletPage = () => {
     console.log(data);
   };
 
+  const downloadFile = async (url: string, fileName: string) => {
+    const response = await fetch(url, {
+      method: 'GET', // *GET, POST, PUT, DELETE, etc.
+      mode: 'cors', // no-cors, *cors, same-origin
+      cache: 'no-cache', // *default, no-cache, reload, force-cache, only-if-cached
+      credentials: 'same-origin', // include, *same-origin, omit
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+        'x-api-key': loggedIn,
+      },
+    });
+    const data = await response.blob();
+
+    const downloadUrl = URL.createObjectURL(data);
+    const a = document.createElement('a');
+    a.href = downloadUrl;
+    a.download = fileName;
+    document.body.appendChild(a);
+    a.click();
+    document.body.removeChild(a);
```

Comment:
Can you check if the link flashes on and off the screen with different browsers (chrome, safari, firefox)?
If so, maybe you can position it off-screen. I don't think you can click a link with JavaScript if display is set to none or visibility false.

Reply:
Test on Chrome, Safari and FireFox: Nothing behaves different.